### DISCUSSION TITLE
Adding custom rules for an end user

### DISCUSSION THREAD

**Discussion by mmolosay** [detekt/detekt#6432] [General]
I have my library in a module `:debounce`. I have sibling module `:lint-rules` with custom rules for my library.
However, those rules are not for the library code itself, but for a code that uses the library.
I want a person who uses my library to see hints and warnings about how to use library better / properly in their IDE, in their code.

I'm having a hard time trying to achieve this. Can anyone help?
Haven't found anything similar in issues and discussions. Could have missed it.

This is my repository: https://github.com/mmolosay/debounce/tree/lint-rules

**Comment by BraisGabin** [comment]
You must publish it and then the final users should add that dependency as any other detekt plugin. That should be enough. To get integration with the IDE they should also configure it on the ide but I never used it so no idea how.

First of all I would focus on get the rule working with the gradle task. Then you can check for the ide integration.

**Reply by mmolosay** [reply]
Is there a way to achieve desired behaviour without forcing user to depend on anything except my library?
Perhaps with a tool other than Detekt?

**Reply by BraisGabin** [reply]
I think that Android lint has that feature enabled by default on android studio. I know that timber (by Jake Wharton) has it implemented.

**Reply by mmolosay** [reply]
Yep, that's the exact behaviour I'm trying to achieve. Seems impossible for regular java library

**Reply by cortinico** [reply]
I don't believe this is easily possible out of the box. Lint achieves this by including a `lint.jar` inside the `.aar` of the library.

detekt instead works by collecting external rules using the `detektPlugin` Gradle Configuration. To achieve what you want, we would have to inspect the `compileOnly` configuration, unzip all the .jar/.aar and look for an arbitrary `detekt.jar` which libraries might include. It seems like a bit of an overkill, especially for a really limited set of libraries.

What's the problem in asking your users to include 2 gradle lines rather than a single one?
